### ISSUE TITLE
`Field` doesn't play well with `TypeAliasType`

### ISSUE THREAD

**Issue by adriangb** [pydantic/pydantic#6352]
```python
from typing import Annotated
from typing_extensions import TypeAliasType

from pydantic import Field, BaseModel

SomeAlias = TypeAliasType('SomeAlias', Annotated[int, Field(description='number')])

class Model(BaseModel):
    foo: Annotated[SomeAlias, Field(title='abc')]


print(Model.model_json_schema())  # description gets dropped
```

Selected Assignee: @hramezani

**Comment by chrisbouchard** [comment]
I can confirm this is still an issue in 2.6.1. The issue seems to be that Pydantic doesn't check if [`TypeAliasType.__value__`](https://docs.python.org/3.12/library/typing.html#typing.TypeAliasType.__value__) would expand to an `Annotated`.

#### With `TypeAliasType`

```python
>>> from pydantic import BaseModel, Field
>>> from typing import Annotated
>>> type MyInt = Annotated[int, Field(default=42)]
>>> class MyModel(BaseModel):
...   my_int: MyInt
... 
>>> MyModel.model_fields
{'my_int': FieldInfo(annotation=MyInt, required=True)}
```

#### With a plain type alias

```python
>>> from pydantic import BaseModel, Field
>>> from typing import Annotated
>>> MyIntAlias = Annotated[int, Field(default=42)]
>>> class MyModelAlias(BaseModel):
...   my_int: MyIntAlias
... 
>>> MyModelAlias.model_fields
{'my_int': FieldInfo(annotation=int, required=False, default=42)}
```

**Comment by MixKlim** [comment]
I'll see what I can do here.

**Comment by thejcannon** [comment]
I've poked at this for more hours than I'm proud of and I think I've narrowed down where the unfortunate behavior loosely is.

I'm still not sure exactly the right fix for it, but I think the problem boils down to:

```python
MyType = TypeAliasType("MyType", Annotated[bool, FieldInfo(description="")])

class Model(BaseModel):
    my_field: MyType
```   

when the Pydantic modeling tries to stamp everything out, the `FieldInfo` js annotations are associated with the _field_ and not the (psuedo-)definition, and therefore the `description` (and friends) get added to the field schema only to be immediately removed once the supporting machinery notices the field is a `$ref`.

So I think the problem is it's kind of it's own type, but one without a dedicated schema generation step. I'm still trying to find the right Je ne sais quoi on the right place to ensure the annotations are applied to the definition and not the field.

**Comment by Viicos** [comment]
Update: This will be fixed in 2.11, however field-specific metadata _won't_ be supported (context available [here](https://github.com/pydantic/pydantic/issues/11467)). By field-specific metadata, I mean any metadata that only makes sense in the context of a model field: aliases, default values, deprecated message, etc.

https://github.com/pydantic/pydantic/pull/11475 updates the documentation on this subject.
